File: A8-110403516.py
"""
Assignment 8
Name: 李倬安
Student Number: 110403516
Course 2022-CE1001-A
"""


def preprocess():
    with open('input.txt', 'r') as file:
        data = list(map(int, file.read().split()))
        file.close()
    return data


def bubble(data, original):
    for i in range(len(data)-1):
        for j in range(len(data)-1-i):
            if data[j] > data[j+1]:
                data[j], data[j+1] = data[j+1], data[j]
        showlist(i+1, data)
    return show_original(original)


def showlist(count, listname):
    print('第', count, '次', listname)


def show_original(listname):
    print('原資料：', listname)


oringinal = preprocess()
bubble(preprocess(), oringinal)

File: A8-110403516.cpp
 /*
 * Assignment 8
 * Name: 李倬安
 * Student Number: 110403516
 * Course 2022-CE1001-A
 */

#include <iostream>
using namespace std;
#include <string>
#define ll long long
using namespace std;
int lcm(int a, int b);
int gcd(int a, int b);
int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    int a, b;
    cin >> a >> b;
    cout << lcm(a, b) << endl;
    return 0;
}

int lcm(int a, int b)
{
    return a * b / gcd(a, b);
}
int gcd(int a, int b)
{
    if (b == 0)
        return a;
    else
        return gcd(b, a % b);
}